swarm.js

```diff
@@ -14,87 +13,69 @@ const CLOSED = Symbol('NETWORK_CLOSED')
 module.exports = opts => new Swarm(opts)
 
 class Swarm extends EventEmitter {
-  constructor (opts) {
-    if (!opts) opts = {}
-
+  constructor (opts = {}) {
     super()
+    const { bootstrap, ephemeral, socket } = opts
+    this.network = guts({
+      bootstrap,
+      discovery: { ephemeral, socket },
```

Comment:
we'll have to implement this in guts btw

Reply:
Ah yes, just move bootstrap to this discovery map, and pass that full map as the options to https://github.com/hyperswarm/guts/blob/master/index.js#L127 (socket is handled internally in guts as self.utp on that line)

Reply:
:+1: